usaspending_api/etl/management/commands/check_es_doc_duplication.py

```diff
@@ -0,0 +1,245 @@
+            "time. The results are then aggregated in Python on the client-side, so it could be memory intensive"
+            "on the client host on which it is run. partition: will run terms aggregations on disjoint "
+            "partitions of the docs, returning --partition-size agg buckets with each call. The aggs being done "
+            "concurrently on the server can cause server straing on CPU, memory, and induce request timeouts",
+        )
+        parser.add_argument(
+            "--es-hostname",
+            type=str,
+            help="The HTTP(S) URL of the Elasticsearch cluster endpoint",
+        )
+        parser.add_argument(
+            "--es-timeout",
+            type=int,
+            help="(default: 300) "
+            "How many seconds the client will wait for the cluster to complete a query before timing out and "
+            "retrying",
+            default=300,
+            metavar="<int>",
+        )
+        parser.add_argument(
+            "--index",
+            type=str,
+            help="Provide name for the index to check. Wildcards and aliases allowed.",
+        )
+        parser.add_argument(
+            "--parallelism",
+            type=int,
+            help="(default: 4, max: 256) "
+            "When using a partition query, number of parallel threads to spin up, each performing an agg query with a "
+            "distinct partition.",
+            default=4,
+            choices=range(1, 257),
+            metavar="{1..256}",
+        )
+        parser.add_argument(
+            "--partition-size",
+            type=int,
+            help="(default: 5000, max: 10000) "
+            "For scroll query, the size of the query result set. For partition query, the size of the aggregation, "
+            "which yields the number of buckets (each for a unique _id) generated "
+            "by each count query. Lower value yields a greater number of queries needing to be run in total; "
+            "higher value yields fewer total queries run. Max is 10,000 (AWS ES Limit).",
+            default=5000,
+            choices=range(1, 10001),
+            metavar="{1..10000}",
+        )
+        parser.add_argument(
+            "--stop-after",
+            type=int,
+            help="When testing, or sampling, halt the checks after this many partitions have been looked at.",
+            metavar="<int>",
+        )
+
+    def handle(self, *args, **options):
+        if options["query_type"] == "scroll":
+            _log.info("Starting to gather doc _ids using 'scroll' query strategy")
+            self.handle_with_scrolling(**options)
+        elif options["query_type"] == "partition":
+            _log.info("Starting to gather doc _ids using 'partition' query strategy")
+            self.handle_with_partitioning(**options)
+        else:
+            raise RuntimeError("Invalid argument provided for required option --query-type.")
```

Comment:
Not worth blocking on, but you can add this check in the argument parser using `choices`: https://docs.python.org/3/library/argparse.html#choices